testsuite/MDAnalysisTests/analysis/test_gnm.py

```diff
@@ -80,6 +81,18 @@ def test_generate_kirchoff(universe):
        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
 
 
+def test_gnm_SVD_fail(universe):
+    gnm = mda.analysis.gnm.GNMAnalysis(universe)
+    # Initilize attributes like self._ts 
+    # necessary for print the warning
+    gnm.run(stop=1)
+
+    with patch.object(np.linalg, "svd") as np_load_mock:
+        np_load_mock.side_effect = np.linalg.LinAlgError
+        with pytest.warns(UserWarning):
+            assert gnm._single_frame() is None
```

Comment:
Either this is True right? We probably don't need an assert here, but if you want it, would it be better to just assert on a zero length results list?

Reply:
You are right. No we don't necessarily need the assert here.